### DISCUSSION TITLE
Disable creality logging to preserve NAND - (MASSIVE XXXGB+ /day writes?)

### DISCUSSION THREAD

**Discussion by dnyank1** [Guilouz/Creality-Helper-Script-Wiki#632] [Features Request]
As some in the community have noted - https://www.reddit.com/r/crealityk1/comments/1c9nqem/massive_amount_of_writes_to_the_internal_emmc/

creality's internal apps aren't well behaved regarding NAND writes. Short of nuking them entirely, I'm desiring a solution to truncate or otherwise disable the printer's logging functions

**Comment by pauloeduardogodoy** [comment]
I read the entire Reddit post and got a little scared... Have been running my K1 Max rooted since day 1......

**Comment by kisslorand** [comment]
In "/usr/data/creality/userdata/log/log_config.json" change <"log_route":1> to <"log_route":0>. 

You can use in SSH the following command:
`sed -i 's/"log_route":1/"log_route":0/' /usr/data/creality/userdata/log/log_config.json`

**Comment by jhstatewide** [comment]
I am so glad I stumbled upon this thread. Seems like a lot of wear to the eMMC for very little gain. Very grateful these printers are rootable and we can make a change like this.

**Comment by imitheo** [comment]
Any chance my klipper forum thread is related to this?
https://klipper.discourse.group/t/shutdown-being-invoked-by-mcu/18472

**Reply by jhstatewide** [reply]
Potentially -- why not disable the logs and see if it resolves your problems? Should be easy enough to do a before/after. Please let us know.

**Reply by mfraser** [reply]
This might also have been the cause of my errors this week. I've now reverted back to
`<"log_route":1>`
Going to see if that fixes my issues.

Is there a reason why we can't change the log level instead?
`"log_level":"LOG_DEFAULT=0, LOG_DEBUG=1, LOG_INFO=2, LOG_WARNING=3, LOG_ERROR=4",`
`"log_level":1`

**Reply by mfraser** [reply]
After changing log_route back to 1 my issues with not being able to use the touch screen after a print had completed are gone.

**Reply by dnyank1** [reply]
@mfraser Maybe I misunderstood what @imitheo is saying here, but I don't think he's made any modifications to the logging behavior of his printer yet. He's reporting other issues. 

Honestly I've been rocking log_route 0 since that bit of knowledge got dropped and... no issues to report. But I also wouldn't see why log behavior would impact the touch screen - are you sure that's not a false correlation somehow?

**Reply by mfraser** [reply]
After I changed to "log_level":0 I was unable to turn the fans down or change temperature on the touch screen and the following print would fail after doing the purge line. At first I didn't know what the issue was, but when that was posted above I changed it back.
Now that it is back as it was everything is working again. Nothing else was changed so it must've been changing that line.
Would changing to LOG_WARNING or LOG_ERROR reduce the writes too?

**Reply by dnyank1** [reply]
You can try for yourself and report back! I was able to verify my printer's write behavior using SSH commands - "iostat -m" and "uptime" specifically. With log route 0 and fairly continuous usage over 2 days 13 hours uptime I have ~1860MB written. I was getting (literally) thousands of megabytes written within only a few minutes of uptime before changing the logging behavior. This seems to be persistent across reboots as well

**Reply by mfraser** [reply]
I think I will, but I still can't understand why setting log_route to 0 is breaking things on my printer. Unless it's because I'm still on 1.3.2.20?
